HW0/P4/P4.py

```diff
@@ -30,47 +30,116 @@
     # Load true trajectory and plot it
     # Normally, this data wouldn't be available in the real world
     #####################
+    x_coords, y_coords,z_coords, v_x, v_y, v_z = np.loadtxt('P4_trajectory.txt', 
+        delimiter=',',unpack=True)
 
-    # ax.plot(x_coords, y_coords, z_coords,
-    #         '--b', label='True trajectory')
+    ax.plot(x_coords, y_coords, z_coords,
+             '--b', label='True trajectory')
 
     #####################
     # Part 2:
     #
     # Read the observation array and plot it (Part 2)
     #####################
-
-    # ax.plot(x_coords, y_coords, z_coords,
-    #         '.g', label='Observed trajectory')
+    x_coords, y_coords, z_coords = np.loadtxt('P4_measurements.txt', delimiter=',',unpack=True)
+    ax.plot(x_coords/rx, y_coords/ry, z_coords/rz,
+             '.g', label='Observed trajectory')
 
     #####################
     # Part 3:
     # Use the initial conditions and propagation matrix for prediction
     #####################
 
-    # A = ?
-    # a = ?
-    # s = ?
+    # given
+    A = np.matrix([
+        [1,0,0,dt,0,0],
+        [0,1,0,0,dt,0],
+        [0,0,1,0,0,dt],
+        [0,0,0,1-c*dt,0,0],
+        [0,0,0,0,1-c*dt,0],
+        [0,0,0,0,0,1-c*dt]])
+    a = np.matrix([[0],[0],[0],[0],[0],[g*dt]])
+    s = np.matrix([[0],[0],[2],[15],[3.5],[4.0]])
 
     # Initial conditions for s0
     # Compute the rest of sk using Eq (1)
 
-    # ax.plot(x_coords, y_coords, z_coords,
-    #         '-k', label='Blind trajectory')
+    # new matrix vectors
+    x_coords2 = [s.item((0,0))]
+    y_coords2 = [s.item((1,0))]
+    z_coords2 = [s.item((2,0))]
+
+    # iterate and update for x,y,z vals
+    prev = s
+    for _ in range(K-1):
```

Comment:
Agreed, didn't know that either! Is _ just treated like any other char (ie is that the same as for i in range(K-1)) or is it different at all?